ext/php5/ddtrace.c

```diff
@@ -1859,11 +1859,16 @@ static const zend_function_entry ddtrace_functions[] = {
     DDTRACE_SUB_NS_FE("Testing\\", trigger_error, arginfo_ddtrace_testing_trigger_error),
     DDTRACE_FE_END};
 
-zend_module_entry ddtrace_module_entry = {
-    STANDARD_MODULE_HEADER,  PHP_DDTRACE_EXTNAME,          ddtrace_functions,      PHP_MINIT(ddtrace),
-    PHP_MSHUTDOWN(ddtrace),  PHP_RINIT(ddtrace),           PHP_RSHUTDOWN(ddtrace), PHP_MINFO(ddtrace),
-    PHP_DDTRACE_VERSION,     PHP_MODULE_GLOBALS(ddtrace),  PHP_GINIT(ddtrace),     NULL,
-    ddtrace_post_deactivate, STANDARD_MODULE_PROPERTIES_EX};
+static const zend_module_dep ddtrace_module_reqs[] = {ZEND_MOD_REQUIRED("json") ZEND_MOD_END};
```

Comment:
This is how we tried to resolve the same issue with the curl extension in #1144. We had to revert it (#1181 & #1209) and resolve the symbol addresses on startup because the extension might be loaded as a shared lib without the headers installed.

Reply:
But, this is for runtime? i.e. PHP will emit a proper warning for us and turn the tracer off then?

Reply:
Yes. But I think this check would be better suited at the call to `zai_json_setup_bindings()`. That way we also catch the unlikely case that the json extension is loaded but the symbols still could not be resolved. (This also makes it consistent with how the [tracer handles this same edge case](https://github.com/DataDog/dd-trace-php/blob/master/ext/php8/handlers_curl.c#L355-L357) with curl).